webcomponents/tool/src/main/frontend/js/tasks/sakai-tasks-create-task.js

```diff
@@ -77,7 +78,12 @@ export class SakaiTasksCreateTask extends SakaiDialogContent {
     this.task.siteId = this.siteId;
     this.task.userId = this.userId;
     this.task.owner = this.task.owner || this.userId;
+    this.task.complete = this.completed;
```

Comment:
I feel that this logic should happen on the backend. The server side code knows if a task has been completed. Also, can a task be 'completed' and 'softDeleted' at the same time. I feel like a task can be both. With this kind of work, frontend stuff, you should try and let your data control the behaviour of your frontend components. If you find yourself adding little bits of logic like this you should stop and have another think about what you are trying to achieve.

Reply:
Hello Adrian, you are absolutely right that the server-side code is responsible for knowing if a task has been completed. However, this information is primarily used to set the initial value of the task widget. When users check the 'completed' checkbox and subsequently click the 'save' button, it becomes necessary to update the properties of the current task and send them to the server. This requires obtaining the current value of the checkbox.

In my recent commit, I have incorporated ```this.shadowRoot.getElementById("complete")``` to retrieve the value of the 'completed' checkbox. I then assign this value to the current task, similar to how we update the description:
```
save() {
  this.task.description = this.shadowRoot.getElementById("description").value
  ...
  // Add this
  const completeEl = this.shadowRoot.getElementById("complete");
  if (completeEl) {
      this.task.complete = completeEl.checked;
    }
  ...
}
```
Furthermore, I have removed the 'complete()' function as it appears unnecessary based on my evaluation.

Reply:
Regarding of the ```softDeleted```, I agree with you that a task cannot be both 'completed' and 'softDeleted' simultaneously. The reason I initially included the code was because it resided within the ```complete()``` function. However, I have now removed those lines to since we might not need it.